src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -621,6 +642,17 @@ public PreferencesViewModel(DynamoViewModel dynamoViewModel)
             SavedChangesTooltip = string.Empty;
 
             this.PropertyChanged += model_PropertyChanged;
+
+            //Expanders settings are stored by session (when dynamo is closed, settings are lost)
+            ExpandersSettings = new List<ExpanderSettings>
+            {
+                new ExpanderSettings{ Name = "PythonExpander", IsExpanded = false, Tab = "Features" },
+                new ExpanderSettings{ Name = "ExperimentalExpander", IsExpanded = false, Tab = "Features" },
+                new ExpanderSettings{ Name = "Styles", IsExpanded = false, Tab = "Visual Settings" },
+                new ExpanderSettings{ Name = "Scale", IsExpanded = false, Tab = "Visual Settings" },
+                new ExpanderSettings{ Name = "Precision", IsExpanded = false, Tab = "Visual Settings" },
+                new ExpanderSettings{ Name = "Display", IsExpanded = false, Tab = "Visual Settings" },
```

Comment:
Do you think we can auto populate this list somehow?

Reply:
@QilongTang I cannot think of any way of auto populate this list but in order to avoid doing this I re-implemented the functionality of recording/loading the current expansion state of each expander in a List<ExpanderSettings>, now instead of recording each one only records the one expanded (only one Expander can be expanded per each tab), so the info for just one is loaded when the Preferences window is opened and just one info is saved when the Expander expansion state is updated (this approach was suggested by @Astul-Betizagasti ).
For implementing this functionality I added a Converter that will be binded in each Expander.IsExpanded as @mjkkirschner  suggested.
This re-implementation allows to do the same functionality for Expanders with less code. Thanks

See commit:
https://github.com/RobertGlobant20/Dynamo/commit/3924722fe2c1c05c43f566ae31d511f41996ee6c

Reply:
Thank you @RobertGlobant20 @Astul-Betizagasti I am fine with the new approach as it reduces some complexity